pkg/ipam/poolallocator/allocator.go

```diff
@@ -37,29 +38,38 @@ type IPPoolAllocator struct {
 	// Name of IP Pool custom resource
 	ipPoolName string
 
-	// crd client to access the pool
+	// crd client to update the pool
 	crdClient crdclientset.Interface
+
+	// pool lister for reading the pool
+	ipPoolLister informers.IPPoolLister
 }
 
 // NewIPPoolAllocator creates an IPPoolAllocator based on the provided IP pool.
-func NewIPPoolAllocator(poolName string, client crdclientset.Interface) (*IPPoolAllocator, error) {
+func NewIPPoolAllocator(poolName string, client crdclientset.Interface, poolLister informers.IPPoolLister) (*IPPoolAllocator, error) {
 
 	// Validate the pool exists
 	// This has an extra roundtrip cost, however this would allow fallback to
 	// default IPAM driver if needed
-	_, err := client.CrdV1alpha2().IPPools().Get(context.TODO(), poolName, metav1.GetOptions{})
+	_, err := poolLister.Get(poolName)
 	if err != nil {
 		return nil, err
 	}
 
 	allocator := &IPPoolAllocator{
-		ipPoolName: poolName,
-		crdClient:  client,
+		ipPoolName:   poolName,
+		crdClient:    client,
+		ipPoolLister: poolLister,
 	}
 
 	return allocator, nil
 }
 
+func (a *IPPoolAllocator) readPool() (*v1alpha2.IPPool, error) {
```

Comment:
Why it is not getPool()?

Reply:
This commit is cherry-picked from #3046 , and this PR relies on #3046 for the IPAM controller part.
@annakhm Could you explain the name design?

Reply:
I can change this to `getPool`